sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/ui/SyncActivity.kt

```diff
@@ -86,17 +90,24 @@ class SyncActivity : DuckDuckGoActivity() {
     @Inject
     lateinit var shareAction: ShareAction
 
-    private val deviceSyncStatusToggleListener: OnCheckedChangeListener = object : OnCheckedChangeListener {
-        override fun onCheckedChanged(
-            buttonView: CompoundButton?,
-            isChecked: Boolean,
-        ) {
-            viewModel.onToggleClicked(isChecked)
+    private val loginFlow = registerForActivityResult(LoginContract()) { resultOk ->
+        if (resultOk) {
+            viewModel.onLoginSuccess()
+        }
+    }
+
+    private val enterCodeLauncher = registerForActivityResult(
+        EnterCodeContract(),
+    ) { resultOk ->
+        if (resultOk) {
+            viewModel.onLoginSuccess()
         }
     }
 
     private val connectFlow = registerForActivityResult(ConnectFlowContract()) { resultOk ->
-        // no-op
+        if (resultOk) {
```

Comment:
I believe this is the bit related to https://app.asana.com/0/0/1205581505958088/f